### DISCUSSION TITLE
vacuum platform selection

### DISCUSSION THREAD

**Discussion by poudenes** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#590] [General]
Hi All,

I have a Roborock S8. For all the sensors and information in use this integration:

https://github.com/humbertogontijo/homeassistant-roborock

Im installed the Lovelace vacuum map card. Looks very nice and clean. But there are some things not working:

The option Rooms in Zone Cleanup, Pin & go work. I can select a room. But when I press the play button on the right of it I get a error message:

Failed to call service Xiaomi_miio/vacuum_clean_segment. Service not found.

When I select "Send_command" in vacuum platform I get this message:
Success

But nothing happen...

Do I use the wrong integration to get the information of the S8 or do I use the wrong vacuum platform?

Oh selecting Mop Mode, Mop intensity and Fan Speed working. A ping sound is giving from the S8.

**Comment by PiotrMachowski** [comment]
You can install the latest beta version, it contains Roborock platform

**Reply by poudenes** [reply]
More then happy to test it!

**Reply by poudenes** [reply]
Have 2.2.0.b1 now. Don't see roborock in the list.

**Reply by PiotrMachowski** [reply]
Clear the cache

**Reply by poudenes** [reply]
Ohh yes.. always the cache issue in combination with HA. worked.. sorry!

**Comment by poudenes** [comment]
offtopic. Do you know why you can select the Mop Mode and Mop Intensity selection. But not when you using the service in develop tool?
Action is same right: service: select.select_option

**Reply by PiotrMachowski** [reply]
You probably do something incorrectly

**Reply by poudenes** [reply]
This is same action as when you select it from pull down menu.
```
service: select.select_option
data:
  option: Intens
target:
  entity_id: select.roborock_s8_mop_intensity
```

Error: Failed to call service select.select_option. Unknown Error

```
Logger: homeassistant.helpers.script.websocket_api_script
Source: components/select/__init__.py:105 
First occurred: 19:36:20 (4 occurrences) 
Last logged: 19:36:50

websocket_api script: Error executing script. Unexpected error for call_service at pos 1: Option Intens not valid for select.roborock_s8_mop_intensity
websocket_api script: Error executing script. Unexpected error for call_service at pos 1: Option Mild not valid for select.roborock_s8_mop_intensity
websocket_api script: Error executing script. Unexpected error for call_service at pos 1: Option Deep not valid for select.roborock_s8_mop_mode
websocket_api script: Error executing script. Unexpected error for call_service at pos 1: Option Standard not valid for select.roborock_s8_mop_mode
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 452, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 685, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1849, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1889, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 226, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 798, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 980, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 838, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/select/__init__.py", line 105, in async_select_option
    raise ValueError(f"Option {option} not valid for {entity.entity_id}")
ValueError: Option Intens not valid for select.roborock_s8_mop_intensity
```

**Reply by PiotrMachowski** [reply]
Service call is ok, but the value is incorrect. You can check possible values in dev tools/states

[![Open your Home Assistant instance and show your state developer tools.](https://my.home-assistant.io/badges/developer_states.svg)](https://my.home-assistant.io/redirect/developer_states/)

**Reply by poudenes** [reply]
I see it... all low letters... its not my day today... haha
